nixos/modules/services/hardware/keyd.nix

```diff
@@ -0,0 +1,68 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  meta = {
+    maintainers = [ "cidkid" ];
+  };
+
+  options = {
+    services.keyd = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          enable keyd daemon for remapping keys
+        '';
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.keyd;
+        defaultText = literalExpression "pkgs.keyd";
+        description = ''
+          KeyD Package to use
+        '';
+      };
+
+      configuration = mkOption {
+        type = types.attrsOf (types.submodule ({ config, options, ... }: {
+          options = {
+            text = mkOption {
```

Comment:
Do you gain anything by making this a submodule instead of just making it a `listOf`? The name of the config file is largely irrelevant, isn't it? Also I'd prefer making this a `lisfOf path`, to have the config in a separate file, but that's just personal preference.

Reply:
Is there any point to allowing multiple configuration files? That sounds like an implementation detail to me...

Additionally, is there any way we can accept a `nix` attribute set for the actual configuration?

Reply:
@dit7ya requested the ability to have multiple config files

Reply:
But what is the *reason* for multiple files? @aanderse's point about passing an attrset is also valid.